iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/relation/citation/CitationRelationExporterIOUtils.java

```diff
@@ -0,0 +1,96 @@
+package eu.dnetlib.iis.wf.export.actionmanager.relation.citation;
+
+import eu.dnetlib.iis.common.java.io.HdfsUtils;
+import eu.dnetlib.iis.common.schemas.ReportEntry;
+import eu.dnetlib.iis.common.spark.avro.AvroDataFrameSupport;
+import eu.dnetlib.iis.common.spark.avro.AvroDatasetSupport;
+import eu.dnetlib.iis.common.utils.RDDUtils;
+import eu.dnetlib.iis.export.schemas.Citations;
+import org.apache.hadoop.io.Text;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.function.PairFunction;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class CitationRelationExporterIOUtils {
```

Comment:
The methods are generic but I'd keep them bound to citation relation export. This is because all the methods use more generic IO method and make them prepared for usage in citation exporting job. The generic methods are themselves pretty simple and are parameterized by schemas and can be reused in other jobs. For example `readCitations()` could be simply replaced by `new AvroDataFrameSupport(spark).read(path, Citations.SCHEMA$)`, which uses a generic method `read()` of `AvroDataFrameSupport`. By wrapping this simple code in `readCitations()` we make it ready for citation export job and also make it easy to test by replacing underlying IO method with a mock.